apps/video/js/video.js

```diff
@@ -114,6 +115,17 @@ document.addEventListener('visibilitychange', function visibilityChange() {
     }
   }
   else {
+    // Bug 1151775
+    // If the app is coming to the foreground before a sharing activity is
+    // complete, restore the video (the video will have been released before
+    // initiating the sharing activity and is only restored when the sharing
+    // activity is completed). When the sharing activity is complete and
+    // restoreVideo is again called, the restoreVideo function takes care of
+    // not restoring the video when it doesn't need to be restored.
+    if (sharingActivity) {
+      videoHardwareReleased = true; // Otherwise restoreVideo will abort
+      restoreVideo();
+    }
```

Comment:
Also: do you need to check here whether we're displaying a video?  If we're displaying the thumbnails, then presumably it isn't necessary to restore the video, right?